src/platform_linux/laio.c

```diff
@@ -91,19 +113,40 @@ io_handle_init(laio_handle         *io,
 
    status = io_setup(cfg->kernel_queue_size, &io->ctx);
    platform_assert(status == 0);
-   if (cfg->flags & O_CREAT) {
+
+   bool is_create = ((cfg->flags & O_CREAT) != 0);
+   if (is_create) {
       io->fd = open(cfg->filename, cfg->flags, cfg->perms);
-      fallocate(io->fd, 0, 0, 128 * 1024);
    } else {
       io->fd = open(cfg->filename, cfg->flags);
    }
-   platform_assert(io->fd != -1);
+   platform_assert((io->fd != -1),
+                   "open() %s '%s' failed, errno=%d: %s\n",
+                   (is_create ? "new" : "existing"),
+                   cfg->filename,
+                   io->fd,
+                   strerror(errno));
+
+   if (is_create) {
```

Comment:
This fixes a latent bug that will trip and makes the assert on L123 moot.

If `open()` on L19 were to fail, `fallocate()` on old L96 will also seg-fault.

Rearranged code to make it a bit more fail-safe.